### PR TITLE
Upgrade Calico to 3.0.1

### PR DESCRIPTION
**Pull request by r7vme**
Not tested yet.

### CODE REVIEW

File: cloud-config/master.yaml.tmpl [hunk 1]
```diff
@@ -381,12 +520,27 @@ write_files:
...
+      scope: Namespaced
+      group: crd.projectcalico.org
+      version: v1
+      names:
+        kind: NetworkPolicy
+        plural: networkpolicies
+        singular: networkpolicy
+
+    ---
+
```
Comment (kopiczko) [inline cloud-config/master.yaml.tmpl]: Is this the storage?

File: cloud-config/master.yaml.tmpl [hunk 2]
```diff
@@ -136,32 +137,145 @@ write_files:
...
+              periodSeconds: 30
+              initialDelaySeconds: 30
+            readinessProbe:
+              httpGet:
+                path: /readiness
+                port: 9098
+              periodSeconds: 10
+
+    ---
+
```
Comment (kopiczko) [inline cloud-config/master.yaml.tmpl]: Do we use Typha in guest clusters?

Comment (r7vme) [reply cloud-config/master.yaml.tmpl]: We still use old calico in Azure guest clusters and need to update manifests too. So Typha will be added.

File: cloud-config/master.yaml.tmpl (hunk 1 above)
Comment (r7vme) [reply cloud-config/master.yaml.tmpl]: Exactly

**Comment by r7vme** [comment]
Applied to gollum.

Additional migration steps are done manually.
1) Delete all Calico CRDs
2) Delete new Calico with `kubectl delete -f /srv/calico-all.yaml`
3) Apply back `kubectl apply -f /srv/calico-all.yaml`

These steps only for Azure (k8s datastore) for AWS and baremetal they probably not needed, as new Calico by default will use etcd v3 (which is empty for Calico)

**Pull request merged**